analyzers/src/SonarAnalyzer.CSharp/Rules/RedundancyInConstructorDestructorDeclaration.cs

```diff
@@ -106,5 +127,9 @@ private static bool TypeHasExactlyOneConstructor(BaseMethodDeclarationSyntax con
 
         private static bool IsConstructorParameterless(BaseMethodDeclarationSyntax constructorDeclaration) =>
             constructorDeclaration.ParameterList.Parameters.Count == 0;
+
+        private static bool ContainsInitializedFieldOrProperty(INamedTypeSymbol symbol) =>
+            symbol.GetMembers().OfType<IFieldSymbol>().Any(f => f.DeclaringSyntaxReferences.Any(d => ((VariableDeclaratorSyntax)d.GetSyntax()).Initializer != null))
+            || symbol.GetMembers().OfType<IPropertySymbol>().Any(p => p.DeclaringSyntaxReferences.Any(d => ((PropertyDeclarationSyntax)d.GetSyntax()).Initializer != null));
```

Comment:
`!= null` should be `is not null`

Reply:
Here we should be able to define an extension method as we do for `SyntaxNodeExtensions.GetIdentifier`, which would get the `Identifier` for both `VariableDeclaratorSyntax` and `PropertyDeclarationSyntax`. 

That allows you to have a single iteration over the members, instead of two. Something like:
```cs
        private static bool ContainsInitializedFieldOrProperty(INamedTypeSymbol symbol) =>
            symbol.GetMembers().Any(f => f.DeclaringSyntaxReferences.Any(d => d.GetSyntax().GetInitializer() is not null))
```

Reply:
That's even better, thanks for the tip!